File: backend/WorkOrderSystem/src/main/java/com/rj/model/pojo/Role.java
package com.rj.model.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("role")
@Schema(description = "角色实体")
public class Role {

    @Schema(description = "主键")
    @TableId(type = IdType.AUTO)
    private Long id;

    @Schema(description = "角色标识:SUBMITTER/REVIEWER/DISPATCHER/HANDLER/ADMIN(代码判断用)")
    private String roleCode;

    @Schema(description = "角色名称:提单人/审核人/派单人/处理人/管理员(展示用)")
    private String roleName;

    @Schema(description = "职责说明")
    private String remark;

    @Schema(description = "创建时间")
    private LocalDateTime createTime;
}

File: frontend/work_order_system/src/constants/role.ts
/**
 * 角色常量
 *
 * 角色码由后端 role.role_code 定义,基本固定不变(见后端 RoleController 的注释:
 * 「角色基本上固定的 不需要修改」),所以中文名在前端也留一份映射做展示。
 */

/** 角色标识 -> 中文名 */
export const ROLE_NAMES: Record<string, string> = {
  SUBMITTER: '提单人',
  REVIEWER: '审核人',
  DISPATCHER: '派单人',
  HANDLER: '处理人',
  ADMIN: '管理员',
}

/** 查角色中文名;未知角色码原样返回,便于暴露数据问题而不是显示成空白 */
export function getRoleName(code: string | null | undefined): string {
  if (!code) return '—'
  return ROLE_NAMES[code] ?? code
}
